crates/amaru-ledger/src/summary/governance.rs

```diff
@@ -145,3 +153,108 @@ impl GovernanceSummary {
+    //
+    // for epochs 163 and 172, with GOV_ACTION_LIFETIME = 6, proposals_activity_periods would equal
+    //   [163, 164, 165, 166, 167, 168, 169, 172, 173, 174, 175, 176, 177, 178]
+    let proposals_activity_periods = proposals
+        .iter()
+        .flat_map(|(_pointer, start)| {
+            (*start..=*start + governance_action_lifetime).collect::<BTreeSet<_>>()
+        })
+        .collect::<BTreeSet<Epoch>>();
+
+    let most_recent_epoch = proposals_activity_periods.iter().max().copied();
+
+    move |registered_at, last_interaction| {
+        // Each epoch with no active proposals increase the mandate by 1
+        let dormant_epochs =
+            BTreeSet::from_iter(last_interaction..=most_recent_epoch.unwrap_or(last_interaction)) // Total period considered for this DRep
+                .difference(&proposals_activity_periods)
+                .count() as u64;
+
+        last_interaction + drep_expiry + dormant_epochs
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use amaru_kernel::network::{Bound, EraParams, Summary};
+    use std::sync::LazyLock;
+    use test_case::test_case;
+
+    const VERSION_9: ProtocolVersion = (9, 0);
+    const VERSION_10: ProtocolVersion = (10, 0);
+
+    static ERA_HISTORY: LazyLock<EraHistory> = LazyLock::new(|| EraHistory {
+        eras: vec![Summary {
+            start: Bound {
+                time_ms: 0,
+                slot: From::from(0),
+                epoch: 0,
+            },
+            end: Bound {
+                time_ms: 1000000,
+                slot: From::from(1000),
+                epoch: 100,
+            },
+            params: EraParams {
+                epoch_size_slots: 10,
+                slot_length: 1000,
+            },
+        }],
+    });
+
+    fn ptr(slot: u64, transaction_index: usize) -> (TransactionPointer, Epoch) {
+        (
+            TransactionPointer {
+                slot,
+                transaction_index,
+            },
+            ERA_HISTORY.slot_to_epoch(Slot::from(slot)).unwrap(),
+        )
+    }
+
+    #[test_case(VERSION_9, 3, 10, vec![ptr(85, 0)], 85, 8 => 18)]
+    #[test_case(VERSION_10, 3, 10, vec![ptr(85, 0)], 85, 8 => 18)]
+    #[test_case(VERSION_9, 3, 10, vec![ptr(85, 0)], 85, 9 => 19)]
+    #[test_case(VERSION_10, 3, 10, vec![ptr(85, 0)], 85, 9 => 19)]
+    #[test_case(VERSION_9, 3, 10, vec![ptr(85, 0), ptr(145, 0)], 55, 9 => 21)]
+    #[test_case(VERSION_9, 3, 10, vec![ptr(85, 0), ptr(145, 0)], 55, 9 => 21)]
+    #[test_case(VERSION_10, 3, 10, vec![ptr(85, 0), ptr(145, 0)], 55, 9 => 21)]
+    #[test_case(VERSION_10, 3, 10, vec![ptr(85, 0), ptr(135, 0), ptr(205, 0)], 135, 13 => 26)]
+    #[test_case(VERSION_10, 3, 10, vec![ptr(85, 0), ptr(135, 0), ptr(205, 0)], 205, 20 => 30)]
+    #[test_case(VERSION_10, 3, 10, vec![ptr(85, 0), ptr(135, 0), ptr(205, 0)], 125, 15 => 28)]
+    #[test_case(VERSION_10, 3, 10, vec![ptr(85, 0), ptr(125, 0), ptr(155, 0)], 85, 13 => 23)]
```

Comment:
Note: those test cases are not _complete_ yet. I realized that the behavior around how to properly test VERSION_9 requires to know slots that contain transactions without _any_ certificates since this is where the dormant epoch counter gets reset on the Haskell implementation. DReps registrations that occurs in an epoch after a transaction with no certificates has been processed shouldn't be impacted by the expiry life extension.